src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEAFULT_SIZE = 8;
+    private static Object[] stack;
+    private static int size = 0;
```

Comment:
The 'size' field should not be static for the same reason as the 'stack' field. Each instance of MateStack should maintain its own size.